### DISCUSSION TITLE
Component API - computedProps

### DISCUSSION THREAD

**Discussion by mq1lv** [quasarframework/quasar#14875] [General - Components / Directives / etc] [answered]
Hi all! 

I'm need help with q-uploader component. What is the right way to access computedProps - files?

Thanks!

**Comment by yusufkandemir** [comment, marked as answer]
https://vuejs.org/guide/essentials/template-refs.html

```html
<template>
  <q-uploader ref="uploaderRef" ... />

  <q-btn label="Log Files" @click="logFiles" />
</template>

<script setup lang="ts">
import type { QUploader } from 'quasar';
import { ref, Ref } from 'vue';

const uploaderRef = ref() as Ref<QUploader>;

function logFiles() {
  console.log(uploaderRef.value.files)
}
</script>
```

**Reply by mq1lv** [reply]
This way I get undefined. I have tried uploaderRef.value.files, uploaderRef.value.queuedFiles ... always undefined. Is there other way?

Accessing methods with uploaderRef.value.methodName() - works.

**Reply by yusufkandemir** [reply]
Make sure you are using `quasar` v2.8.0+
